src/util/schnorr.rs

```diff
@@ -52,29 +52,59 @@ impl str::FromStr for TweakedPublicKey {
     }
 }
 
-/// A trait for tweaking Schnorr public keys
+/// Untweaked BIP-340 key pair
+pub type UntweakedKeyPair = KeyPair;
+
+/// Tweaked BIP-340 key pair
+#[derive(Clone)]
+#[cfg_attr(feature = "std", derive(Debug))]
+// TODO: Add other derives once secp256k1 v0.21.3 released
+pub struct TweakedKeyPair(KeyPair);
+
+/// A trait for tweaking BIP340 key types (x-only public keys and key pairs).
 pub trait TapTweak {
-    /// Tweaks an untweaked public key given an untweaked key and optional script tree merkle root.
-    ///
-    /// This is done by using the equation Q = P + H(P|c)G, where
-    ///  * Q is the tweaked key
-    ///  * P is the internal key
+    /// Tweaked key type with optional auxiliary information
+    type TweakedAux;
+    /// Tweaked key type
+    type TweakedKey;
+
+    /// Tweaks an untweaked key with corresponding public key value and optional script tree merkle
+    /// root. For the [`KeyPair`] type this also tweaks the private key in the pair.
+    /// 
```

Comment:
In 73f7e96ab15308afc4c1674068044edc38bd60db:

There are trailing spaces here and on two other lines in this file.